### PR TITLE
validate CHANGELOG.md is well-formed

### PR DESCRIPTION
**Pull request by andrewhsu**
**- What I did**

Added validation script to check `CHANGELOG.md` is well-formed so RPM builds can process the changelog.

Also fixed one line in `CHANGELOG.md` so that all version lines are consistent.

**- How I did it**

This script just checks the version line and makes sure it is of format similar to:

```
## 1.12.3 (2016-10-26)
```

**- How to verify it**

```
$ hack/validate/default
Congratulations!  All commits are properly signed with the DCO!
Congratulations!  All Go source files are properly formatted.
Congratulations!  All Go source files have been linted.
Congratulations!  "./pkg/..." is safely isolated from internal code.
No api/types/ or api/swagger.yaml changes in diff.
Congratulations!  No testing.T found.
Congratulations!  All toml source files changed here have valid syntax.
Congratulations!  All Go source files have been vetted.
Congratulations!  Changelog CHANGELOG.md is well-formed.
```

I also added processing of the first argument to the script so you can override the changelog file to validate, e.g.
```
$ hack/validate/changelog-well-formed MALFORMED-CHANGELOG.md 
Malformed changelog MALFORMED-CHANGELOG.md line "## 1.13.0"
$ echo $?
1
```

**- Description for the changelog**

N/A

**- A picture of a cute animal (not mandatory but encouraged)**

🐋 

Signed-off-by: Andrew Hsu <andrewhsu@docker.com>

### CODE REVIEW

**Comment by thaJeztah** [comment]
SGTM, nice!

ping @tianon perhaps you want to have a look?

**Review by vdemeester — APPROVED** [review]
LGTM 🐻

**Comment by vieux** [comment]
very cool LGTM

File: hack/validate/changelog-well-formed [hunk 1]
```diff
@@ -0,0 +1,25 @@
+#!/bin/bash
+
+changelogFile=${1:-CHANGELOG.md}
+
+if [ ! -r "$changelogFile" ]; then
+  echo "Unable to read file $changelogFile" >&2
+  false
```
Comment (AkihiroSuda) [inline hack/validate/changelog-well-formed]: don't we need `exit` here? (or `set -e`?)

Comment (andrewhsu) [inline hack/validate/changelog-well-formed]: `false` is a bash builtin command that will exit with value 1: http://www.tldp.org/LDP/abs/html/internal.html#FALSEREF

Comment (AkihiroSuda) [inline hack/validate/changelog-well-formed]: `false exit` with status 1, but `changelog-well-formed` does not exit here

https://goo.gl/kweyAV

Comment (andrewhsu) [inline hack/validate/changelog-well-formed]: that's a cool website!

i've changed `false` to `exit`

Comment (thaJeztah) [inline hack/validate/changelog-well-formed]: > that's a cool website!

It runs on docker :smile:

/cc @codingground

**Review by AkihiroSuda — APPROVED** [review]
LGTM

**Pull request merged**